socorro/external/postgresql/signature_summary.py

```diff
@@ -109,6 +110,9 @@ def get(self, **kwargs):
         params['version'] = tuple(versions)
 
         all_results = {}
+        assert isinstance(params['start_date'], datetime.date)
+        assert isinstance(params['end_date'], datetime.date)
```

Comment:
The reason for this added verbosity is to future foot-gun-proof us. Later this month @twobraids and I are planning to rewrite large parts of the implementation classes to use a new annotation system. In particular so that the webapp can scan and auto-generate the API documentation. Then we're going to get rid of the `filters = [` lists found in places like this. And since a `datetime.datetime` is "convertable" to `datetime.date` we must let this happen on the psycopg level because it goes horribly wrong. 

So by putting these asserts in we can assure that we don't accidentally allow any `datetime.datetime` instances in as parameters.